### PR TITLE
fix: fs.readFile does not exist in browser

### PR DESCRIPTION
**Pull request by bcoe**
Fixes: https://github.com/googleapis/google-api-nodejs-client/issues/1614

Folks are bumping into an issue when trying to use google-api-nodejs-client in conjunction with React, this is because `fs.readFile` does not exist in the browser environment, and `pify` gets passed `undefined`.

This fix checks to see if `fs.readFile` is defined, if it is not it replaces `readFile` with a function that will throw if an attempt is made to load credentials using the `keyFile` option.

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -404,6 +404,27 @@ describe('.getToken()', () => {
...
+       // exist. This is the same as when running in the browser.
+       delete require.cache[require.resolve('../src')];
+       delete require.cache[require.resolve('fs')];
+       const fs = require('fs');
+       delete fs.readFile;
+       const {GoogleToken} = require('../src');
+
+       let message;
+       try {
+         const gtoken = new GoogleToken(TESTDATA_KEYFILEJSON);
```
Comment (bcoe) [inline test/index.ts]: note `gtoken.getCredentials` is async, `assert.throw` does not seem to work with async functions?

**Comment by bcoe** [comment]
note I've tested with `create-react-app` and at least confirmed that it boots without an exception now.

**Comment by bcoe** [comment]
`create-react-app`, which seems to be the tool a lot of our users are bootstrapping React apps, seems to obfuscate much of the bundling process -- which is awesome, compared to the state of the world two years ago!

However, from our user's point of view, they simply add:

```
const { google } = require('googleapis')
```

I think our best bet to ensure that `googleapis` (and eventually `grpc-js` libraries) don't break in the browser, would be differing platform specific operations, like:

```
pify(fs.readFile)
``` 

Happen later, only when these operations are required, closer to run time; another hammer we have is detecting whether `window` is defined -- which we could potentially use to fail earlier.

Really curious about other people's thoughts, would be awesome to get some tests around this in our core libraries, to make sure we're future-proofing the ability to run our modules in the browser.

File: test/index.ts (hunk 1 above)
Comment (JustinBeckwith) [reply test/index.ts]: That is very correct.  For that you need to use `assert.rejects`.

**Pull request merged**